staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -87,6 +89,23 @@ func IsWebSocketRequest(req *http.Request) bool {
 	return httpstream.IsUpgradeRequest(req)
 }
 
+// IsWebSocketRequestWithStreamCloseProtocol returns true if the request contains headers
+// identifying that it is requesting a websocket upgrade with a remotecommand protocol
+// version that supports the "CLOSE" signal; false otherwise.
+func IsWebSocketRequestWithStreamCloseProtocol(req *http.Request) bool {
+	if !IsWebSocketRequest(req) {
+		return false
+	}
+	for _, requestedProtocols := range req.Header[WebSocketProtocolHeader] {
```

Comment:
```suggestion
	for _, requestedProtocols := range req.Header.Get(WebSocketProtocolHeader]) {
```

https://pkg.go.dev/net/http#Header.Get

header fields are insensitive https://datatracker.ietf.org/doc/html/rfc2616#section-4.2 and Get already implements it IIRC

Reply:
yeah, it seems so https://github.com/kubernetes/kubernetes/pull/119186#discussion_r1318623699 , the libraries uses .Get

Reply:
Fixed.